core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -952,6 +952,9 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
   final def provideSome[R0](f: R0 => R)(implicit ev: NeedsEnv[R]): ZIO[R0, E, A] =
     ZIO.accessM(r0 => self.provide(f(r0)))
 
+  final def provideSomeLayer[R0 <: Has[_]]: ZIO.ProvideSomeLayer[R0, R, E, A] =
```

Comment:
If this works well, would be nice to add to the usual suspects (`Spec`, `ZManaged`, `ZStream`).

Reply:
Yes I agree. One other thing that might be worth thinking about is our nomenclature for `provide` methods. I think we could potentially implement a similar mechanic for, say providing a managed, by converting the managed to a layer and calling `provideSomeLayer`, with the appropriate implicit evidence. So we could conceptually have three varieties of "provide", (1) provide the whole environment, (2) providing part of the environment by providing something that goes from a part to the whole environment (e.g. existing `provideSome`), and (3) providing part of the environment and leaving the remainder (e.g. new `provideSomeLayer`. And you could have variants for each different data type being provided. Maybe `provideSome` could be use for (3) and we find another name for (2)? Or do you think this should exclusively be a mechanic for layers?

Reply:
I do think it's worth rethinking this right now, before RC18.

One could even get away from `provide` for ZLayer, favoring maybe:

 - `interpret` / `interpretSome` ("interpret from one layer to another")
 - `translate` / `translate` ("translate from one layer to another")
 - `shift` / `shiftSome` ("shift from one layer to another")

Or we could continue running with `provide` and just try to emphasize the primacy of ZLayer.

Now providing method to convert `ZManaged` into a `ZLayer` sounds good and could let us delete the `*Managed` variants without too much fuss.